File: src/Web/Pages/Todo/Todo.cs
using Web.Models;

namespace Web.Pages.Todo;

public partial class Todo
{
    private string newTask = "";

    private List<TodoList> todoLists = new()
    {
        new()
        {
            Title = "Todo List 1",
            TodoItems = new List<TodoItem>
            {
                new()
                {
                    Note = "Tasl 1",
                    Done = true
                },
                new()
                {
                    Note = "Tasl 1",
                    Done = true
                }
            }
        },
        new()
        {
            Title = "Todo List 1",
            TodoItems = new List<TodoItem>
            {
                new()
                {
                    Note = "Tasl 1",
                    Done = false
                },
                new()
                {
                    Note = "Tasl 1",
                    Done = true
                }
            }
        }
    };
}

File: src/Web/src/app/core/models/todo.ts
export interface TodoList {
  id: string,
  title: string,
  todoItems: TodoItem[],
  userId: string,
  createdAt: Date,
  isTodayTodoList: boolean
}

export interface TodoItem {
  id: string,
  note: string,
  done: boolean,
  todoListId: string,
  priority: Priority,
  deadline: string | null,
  createdAt: Date
}

export interface RemoveTodoListResponse {
  message: string
}

export interface CreateTodoListRequest {
  title: string
}

export interface CreateTodoItemRequest {
  note: string,
  todoListId: string,
  priority: Priority,
  deadline: string | null,
}

export interface GetTodoListsResponse {
  todoLists: TodoList[]
}

export enum Priority {
  Low = 0,
  Medium = 1,
  High = 2
}
